File: front/src/types/taxInvoiceReceipt.ts
// Tax Invoice Receipt types - חשבונית מס-קבלה
// מסמך שמשלב יחד חשבונית מס וקבלה, מתאים לעסקים שמקבלים תשלום מיידי

export const TaxInvoiceReceiptStatus = {
  Paid: 1,      // שולם
  Cancelled: 2  // בוטל
} as const;

export type TaxInvoiceReceiptStatus = typeof TaxInvoiceReceiptStatus[keyof typeof TaxInvoiceReceiptStatus];

export interface TaxInvoiceReceiptLine {
  id: number;
  itemId: number;
  itemName: string;
  itemSku: string;
  itemUnit: string;
  quantity: number;
  unitPrice: number;
  discountPercent: number;
  discountAmount: number;
  vatRate: number;
  lineSubTotal: number;
  lineVatAmount: number;
  lineTotalAmount: number;
}

export interface TaxInvoiceReceipt {
  id: number;
  customerId: number;
  customerName: string;
  documentNumber: string;
  documentDate: string;
  status: TaxInvoiceReceiptStatus;
  statusDisplayName: string;
  subTotal: number;
  vatAmount: number;
  totalAmount: number;
  paymentMethod: string;
  referenceNumber?: string;
  currency: string;
  exchangeRate?: number;
  notes?: string;
  createdAt: string;
  lines: TaxInvoiceReceiptLine[];
}

export interface TaxInvoiceReceiptListItem {
  id: number;
  documentNumber: string;
  documentDate: string;
  customerName: string;
  status: TaxInvoiceReceiptStatus;
  statusDisplayName: string;
  totalAmount: number;
  paymentMethod: string;
  currency: string;
}

export interface CreateTaxInvoiceReceiptLine {
  itemId: number;
  quantity: number;
  unitPrice: number;
  discountPercent?: number;
  discountAmount?: number;
  vatRate?: number;
}

export interface CreateTaxInvoiceReceipt {
  customerId: number;
  documentDate: string;
  paymentMethod: string;
  referenceNumber?: string;
  currency?: string;
  exchangeRate?: number;
  notes?: string;
  lines: CreateTaxInvoiceReceiptLine[];
}

export interface UpdateTaxInvoiceReceipt {
  paymentMethod: string;
  referenceNumber?: string;
  notes?: string;
}

export interface TaxInvoiceReceiptFilter {
  documentNumber?: string;
  fromDate?: string;
  toDate?: string;
  customerId?: number;
  status?: TaxInvoiceReceiptStatus;
  paymentMethod?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDescending?: boolean;
}

export interface TaxInvoiceReceiptResponse {
  items: TaxInvoiceReceiptListItem[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export const TAX_INVOICE_RECEIPT_STATUS_LABELS: Record<TaxInvoiceReceiptStatus, string> = {
  [TaxInvoiceReceiptStatus.Paid]: 'שולם',
  [TaxInvoiceReceiptStatus.Cancelled]: 'בוטל'
};

export const PAYMENT_METHODS = [
  'מזומן',
  'כרטיס אשראי',
  'העברה בנקאית',
  'צ\'ק',
  'דיגיטלי'
] as const;

export type PaymentMethod = typeof PAYMENT_METHODS[number];

File: backend/Models/Sales/TaxInvoiceReceipt.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Models.Core;
using backend.Models.Inventory;

namespace backend.Models.Sales;

/// <summary>
/// Tax Invoice Receipt status
/// חשבונית מס-קבלה - מסמך שמשלב יחד חשבונית מס וקבלה
/// מתאים לעסקים שמקבלים תשלום מיידי
/// </summary>
public enum TaxInvoiceReceiptStatus
{
    Paid = 1,        // שולם - המסמך נוצר רק לאחר קבלת התשלום
    Cancelled = 2    // בוטל
}

/// <summary>
/// Tax Invoice Receipt - משלב חשבונית מס וקבלה יחד
/// מסמך שנוצר לאחר קבלת התשלום, מתאים לעסקים שמקבלים תשלום מיידי
/// כמו חנויות קמעונאיות, מסעדות, וכו'
/// </summary>
public class TaxInvoiceReceipt : TenantEntity
{
    [Required]
    public int CustomerId { get; set; }

    /// <summary>
    /// מספר חשבונית מס-קבלה (מספור רציף שנתי)
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string DocumentNumber { get; set; } = string.Empty;

    /// <summary>
    /// תאריך ההפקה והתשלום
    /// </summary>
    [Required]
    public DateTime DocumentDate { get; set; }

    /// <summary>
    /// סטטוס המסמך
    /// </summary>
    [Required]
    public TaxInvoiceReceiptStatus Status { get; set; } = TaxInvoiceReceiptStatus.Paid;

    /// <summary>
    /// סכום כולל לפני מע"מ
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal SubTotal { get; set; }

    /// <summary>
    /// סכום מע"מ
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal VatAmount { get; set; }

    /// <summary>
    /// סכום כולל כולל מע"מ
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal TotalAmount { get; set; }

    /// <summary>
    /// אופן התשלום (מזומן, כרטיס אשראי, העברה בנקאית, וכו')
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string PaymentMethod { get; set; } = string.Empty;

    /// <summary>
    /// מספר אסמכתא (מספר עסקה, מספר צ'ק, וכו')
    /// </summary>
    [MaxLength(100)]
    public string? ReferenceNumber { get; set; }

    /// <summary>
    /// מטבע
    /// </summary>
    [Required]
    [MaxLength(3)]
    public string Currency { get; set; } = "ILS";

    /// <summary>
    /// שער החליפין (אם לא ILS)
    /// </summary>
    [Column(TypeName = "decimal(18,6)")]
    public decimal? ExchangeRate { get; set; }

    /// <summary>
    /// הערות
    /// </summary>
    [MaxLength(1000)]
    public string? Notes { get; set; }

    // Navigation properties
    [ForeignKey("CustomerId")]
    public virtual Customer Customer { get; set; } = null!;

    /// <summary>
    /// פריטי החשבונית מס-קבלה
    /// </summary>
    public virtual ICollection<TaxInvoiceReceiptLine> Lines { get; set; } = new List<TaxInvoiceReceiptLine>();
}

/// <summary>
/// שורת פריט בחשבונית מס-קבלה
/// </summary>
public class TaxInvoiceReceiptLine : TenantEntity
{
    [Required]
    public int TaxInvoiceReceiptId { get; set; }

    [Required]
    public int ItemId { get; set; }

    /// <summary>
    /// כמות
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,3)")]
    public decimal Quantity { get; set; }

    /// <summary>
    /// מחיר יחידה לפני מע"מ
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal UnitPrice { get; set; }

    /// <summary>
    /// אחוז הנחה
    /// </summary>
    [Column(TypeName = "decimal(5,2)")]
    public decimal DiscountPercent { get; set; } = 0;

    /// <summary>
    /// סכום הנחה
    /// </summary>
    [Column(TypeName = "decimal(18,2)")]
    public decimal DiscountAmount { get; set; } = 0;

    /// <summary>
    /// אחוז מע"מ
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(5,2)")]
    public decimal VatRate { get; set; } = 17; // Israeli standard VAT rate

    /// <summary>
    /// סכום שורה לפני מע"מ (לאחר הנחה)
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal LineSubTotal { get; set; }

    /// <summary>
    /// סכום מע"מ לשורה
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal LineVatAmount { get; set; }

    /// <summary>
    /// סכום שורה כולל מע"מ
    /// </summary>
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal LineTotalAmount { get; set; }

    // Navigation properties
    [ForeignKey("TaxInvoiceReceiptId")]
    public virtual TaxInvoiceReceipt TaxInvoiceReceipt { get; set; } = null!;

    [ForeignKey("ItemId")]
    public virtual Item Item { get; set; } = null!;
}
